### PR TITLE
Feature: Add a healthcheck to the Dockerfile

### PR DESCRIPTION
**Pull request by omBratteng**
This feature enables checking the health of the application.
It checks that it can connect to the MongoDB, frontend is reachable, and checks the health of the [apollo server](https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/)

### CODE REVIEW

File: healthcheck.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			}
+		})
+		req.on('error', reject)
+		req.end()
+	})
+}
+
+const checkAPI = (url) => {
+	return new Promise((resolve, reject) => {
+		const req = http.request(url, (res) => {
```
Comment (electerious) [inline healthcheck.js:35]: Ackee contains the `node-fetch` package. We could use it here, too.

Comment (electerious) [reply healthcheck.js:35]: + async/await instead of `new Promise` to keep it in sync with the rest of Ackee.

**Review by electerious — CHANGES_REQUESTED** [review]
Great addition! Thanks a lot!

File: healthcheck.js (hunk 1 above)
Comment (omBratteng) [reply healthcheck.js:35]: Ah, I didn't notice. I'll replace it

File: healthcheck.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+const checkHTTP = async (url) => {
+	return await fetch(url)
+		.then((res) => {
+			if (res.status < 200 || res.status >= 300) {
+				new Error(`Status Code: ${ res.status }`)
+			}
+
+			return res.text() // read the body so it can be garbage collected
```
Comment (electerious) [inline healthcheck.js:26]: Curious about the "read the body so it can be garbage collected". Is that really the case?

File: healthcheck.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+if (dbUrl == null) {
+	signale.fatal('MongoDB connection URI missing in environment')
+	process.exit(1)
+}
+
+const checkHTTP = async (url) => {
+	return await fetch(url)
+		.then((res) => {
+			if (res.status < 200 || res.status >= 300) {
```
Comment (electerious) [inline healthcheck.js:22]: Small tip from my side: res.ok is a boolean and does exactly this check :)

File: healthcheck.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+const dbUrl = process.env.ACKEE_MONGODB || process.env.MONGODB_URI
+const serverUrl = `http://localhost:${ port }`
+
+if (dbUrl == null) {
+	signale.fatal('MongoDB connection URI missing in environment')
+	process.exit(1)
+}
+
+const checkHTTP = async (url) => {
+	return await fetch(url)
```
Comment (electerious) [inline healthcheck.js:20]: Small tip from my side: No need to await here. You aren't doing anything with the result in the `checkHTTP ` function, it's therefore fine to just return the result without awaiting it :)

**Comment by electerious** [comment]
Thanks 🙌

File: healthcheck.js (hunk 4 above)
Comment (omBratteng) [reply healthcheck.js:20]: That is true, think I left it there, in case future expansion

File: healthcheck.js (hunk 3 above)
Comment (omBratteng) [reply healthcheck.js:22]: I was unaware of `res.ok`, feel free to replace it.

File: healthcheck.js (hunk 2 above)
Comment (omBratteng) [reply healthcheck.js:26]: At least a couple of years ago, it was. https://github.com/node-fetch/node-fetch/issues/83
I keep it in just to be on the safe side. Usually could just use a `HEAD` request, but the server isn't listened for `HEAD`

**Pull request merged**